tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/async/JaxRsServerAsyncTest.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.eclipse.microprofile.telemetry.tracing.tck.async;
+
+import static org.eclipse.microprofile.telemetry.tracing.tck.async.JaxRsServerAsyncTestEndpoint.BAGGAGE_VALUE_ATTR;
+import static org.testng.Assert.assertEquals;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.telemetry.tracing.tck.ConfigAsset;
+import org.eclipse.microprofile.telemetry.tracing.tck.TestLibraries;
+import org.eclipse.microprofile.telemetry.tracing.tck.exporter.InMemorySpanExporter;
+import org.eclipse.microprofile.telemetry.tracing.tck.exporter.InMemorySpanExporterProvider;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterProvider;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import jakarta.inject.Inject;
+import jakarta.servlet.http.HttpServletRequest;
+
+class JaxRsServerAsyncTest extends Arquillian {
+
+    @Deployment
+    public static WebArchive createDeployment() {
+
+        ConfigAsset config = new ConfigAsset()
+                .add("otel.bsp.schedule.delay", "100")
+                .add("otel.sdk.disabled", "false")
+                .add("otel.traces.exporter", "in-memory");
+
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(InMemorySpanExporter.class, InMemorySpanExporterProvider.class, HttpServletRequest.class,
+                        JaxRsServerAsyncTestEndpointClient.class, JaxRsServerAsyncTestEndpoint.class)
+                .addAsLibrary(TestLibraries.AWAITILITY_LIB)
+                .addAsServiceProvider(ConfigurableSpanExporterProvider.class, InMemorySpanExporterProvider.class)
+                .addAsResource(config, "META-INF/microprofile-config.properties")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    private static final String TEST_VALUE = "test.value";
+
+    @Inject
+    private Tracer tracer;
+
+    @Inject
+    private InMemorySpanExporter spanExporter;
+
+    @Inject
+    private HttpServletRequest request;
```

Comment:
We shouldn't use the servlet API since it's not part of MicroProfile.

In a test class we should be able to use `@ArquillianResource URL url;`, in a JAX-RS method we should be able to pass `@Context UriInfo uriInfo` as a parameter.